freedom-zephyrdtsfixup-generator.c++

```diff
@@ -0,0 +1,145 @@
+/* Copyright 2018 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include "fdt.h++"
+#include "libfdt.h++"
+#include <fstream>
+#include <iostream>
+#include <string>
+
+using std::cerr;
+using std::endl;
+using std::fstream;
+using std::string;
+
+static void show_usage(string name)
+{
+  std::cerr << "Usage: " << name << " <option(s)>\n"
+	    << "Options:\n"
+	    << "\t-h,--help\t\t\tShow this help message\n"
+	    << "\t-d,--dtb <eg. xxx.dtb>\t\tSpecify fullpath to the DTB file\n"
+	    << "\t-o,--output <eg. dts.fixup>\t\tGenerate Zephyr RTOS DTS fixup file\n"
+	    << endl;
+}
+
+static void write_config_file(const fdt &dtb, fstream &os)
+{
+  os << "/* Copyright (c) 2018 SiFive Inc." << endl;
+  os << " * SPDX-License-Identifier: Apache-2.0 */" << endl;
```

Comment:
The Linux guys told be that SPDX headers must be structured as //-style comments, but I don't know if that applies to Zephyr as well.  I also like adding a "autogenerated by X" header to anything that comes out of a tool and is then checked in, just so people know it's not a human source file.

Reply:
Zephyr has the SPDX headers within the /*...*/ style comments all over the source tree, so this should be fine for them.

I'll add the autogenerated header.